### PR TITLE
feat(wallet-connect): support `safe_setSettings`

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves

The support of `safe_setSettings` in the WalletConnect app.

## How this PR fixes it

The dependencies have been updated, permitting the `safe_setSetting` RPC call.

## How to test it

Prerequisites:
- Open the https://github.com/safe-global/web-core/pull/1162 deployment.
- Run the [EIP-1271 test DApp](https://github.com/5afe/eip-1271-dapp) locally.

Method:
1. Open the WC app in the aforementioned deployment.
2. Connect to the aforementioned test DApp.
3. Toggle off-chain signing and attempt to sign a message.
4. Observe the off-chain signing modal.
5. Sign the message.
6. Observe the signed message in the "Signatures" tab of the transaction list.

### CODE REVIEW

File: apps/wallet-connect/src/hooks/useApps.ts [hunk 1]
```diff
@@ -44,9 +44,13 @@ export function useApps(): UseAppsResponse {
 
   const findSafeApp = useCallback(
     (url: string): SafeAppData | undefined => {
-      let { hostname } = new URL(url)
+      try {
+        const { hostname } = new URL(url)
```
Comment (iamacook) [inline apps/wallet-connect/src/hooks/useApps.ts:48]: `URL` can throw.

File: cypress/e2e/tx-builder/tx-builder.spec.cy.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ describe('Testing Tx-builder safe app', { defaultCommandTimeout: 12000 }, () =>
     cy.findByRole('region').should('exist')
     cy.findByText('test Address Value').should('exist')
     cy.findByText('newValue(address):').should('exist')
-    cy.findAllByText('0x49d4...26A6').should('have.length', 2)
+    cy.findAllByText('0x49d4...26A6').should('have.length', 1)
```
Comment (iamacook) [inline cypress/e2e/tx-builder/tx-builder.spec.cy.js:47]: The recent `safe-react-component` implementation in `web-core` broke these tests (see CI on `development`).

**Review by mikhailxyz — APPROVED** [review]
splendid

**Comment by iamacook** [comment]
Closing in favour of https://github.com/safe-global/safe-react-apps/pull/671.

**Pull request closed without merging**